packages/flame/lib/src/sprite_batch.dart

```diff
@@ -263,7 +274,7 @@ class SpriteBatch {
       color: color ?? defaultColor,
     );
 
-    if (flip && useAtlas && !_hasFlips) {
+    if (_applyFlip && flip && useAtlas && !_hasFlips) {
```

Comment:
Should this produce an error instead? Otherwise the user might get confused why their image is not flipped even though they explicitly passed `flip = true`.

Reply:
That's a good point.
But there is a problem with the flip when only `useAtlas = true`. So rendering by `BatchItem & flip` should work fine. Probably I have to write some docs on it.

Reply:
Maybe there is a better way. Will see it deeper tomorrow.

Reply:
I decided to remove `allowFlip` from the spriteBatch. And will move it to `FlameTileLayer` and here's the reason.
SpriteBatch has not only using useAtlas and also a series of canvas ops when the `useAtlas == false`. But This PR and all the issues from flip basically originated from `drawAtlas`. So it is not necessary to prevent flips in that case.
But then, the question mark may come. Should we also prevent flips when not using `drawAtlas`? 
To solve this ambiguity, some kind of `allowAtlasFlip` and `allowXXXFlip` should come. But it's too complicated and hard to understand and also some properties may inconsistent. 
So I think the best option is removing flips before transforms are added to the SpriteBatch when `allowFlip = false`.

I am trying to decide whether `allowFlip` to change `ignoreFlip` or not. `allowFlip` may forbid flip in the creation of `TiledComponent` if a flip exists. `ignoreFlip` just ignores the flip.
Since it's opt-out, probably the users are conscious of what they're doing.
So I think `ignoreFlip` is better.

Reply:
How do you think?

Reply:
Yeah, I like `ignoreFlip`.
As for changes to `SpriteBatch`, perhaps @spydon can advise?

Reply:
There are no changes to `SpriteBatch` anymore?

Reply:
Yes, there are no changes to SpriteBatch. It moved to TileLayer.